### PR TITLE
Add support for ALPN ("successor" to NPN)

### PR DESCRIPTION
**Pull request by jmesmon**
Implementation and interface is very similar, tests I've created are identical.
It's possible we might be able to reduce duplication between the NPN and ALPN code & tests at some point.

### CODE REVIEW

File: openssl/src/ssl/mod.rs [hunk 1]
```diff
@@ -554,6 +571,29 @@ impl SslContext {
             ffi::SSL_CTX_set_next_protos_advertised_cb(self.ctx, raw_next_protos_advertise_cb, ptr::null_mut());
         }
     }
+
+    /// Set the protocols to be used during ALPN (application layer protocol negotiation).
+    /// If this is a server, these are the protocols we report to the client.
```
Comment (sfackler) [inline openssl/src/ssl/mod.rs]: Can you add a sentence saying that this requires the `alpn` feature?

**Comment by sfackler** [comment]
Needs a rebase and a sentence added to one of the doc blocks, but looks good to me other than that, thanks!

File: openssl/src/ssl/mod.rs (hunk 1 above)
Comment (jmesmon) [inline openssl/src/ssl/mod.rs]: Done

**Comment by jmesmon** [comment]
I've updated with the comment requested & a rebase

**Comment by sfackler** [comment]
Thanks!

**Pull request merged**